### PR TITLE
fix(editor): editor resizing for window resize, md-tabs, etc

### LINKED ISSUE #11: Multiple editors in a page throws error
Hi,
When I try loading multiple editors in a tab view, the second editor throws error.
```HTML
<md-tab label="JS">
     <td-code-editor style="height: 200px" theme="vs" flex language="javascript" [(ngModel)]="jsCode" id="js"></td-code-editor>  
</md-tab>
<md-tab label="HTML">
     <td-code-editor style="height: 200px" theme="vs" flex language="HTML" [(ngModel)]="html" id="html"></td-code-editor>  
</md-tab>

```
error 
```
loader.js:7 Duplicate definition of module '===anonymous1==='
  | s.defineModule | @ | loader.js:7
-- | -- | -- | --
  | t | @ | loader.js:7
  | onGotAmdLoader | @ | code-editor.component.ts:508
  | webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask | @ | zone.js:424
  | onInvokeTask | @ | core.es5.js:3881
  | webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask | @ | zone.js:423
  | webpackJsonp.../../../../zone.js/dist/zone.js.Zone.runTask | @ | zone.js:191
  | ZoneTask.invoke | @ | zone.js:486
```

How to solve this problem, please help

### PR DESCRIPTION
**Pull request by jeremysmartt**
Exposing automaticLayout method that via setInterval that constantly probes for the container's size. Also a layout() method that manually instructs the editor to remeasure its container

closes #11

Test Steps
-[ ] Checkout branch
-[ ] npm run build
-[ ] cd deploy/code-editor
-[ ] npm pack (prints out a tgz file created)
-[ ] pwd (prints out your < path > you are at)
-[ ] Go to Covalent repo
-[ ] rm -rf node_modules/`@covalent`/code-editor
-[ ] npm install < path > / < tgz file created >
-[ ] edit the demo for code-editor.component.html and add the following:
```html
    <md-tab-group>
      <md-tab label="app code">
        <td-code-editor automaticLayout="true" style="height: 200px" theme="vs" flex language="javascript" [(ngModel)]='frontEndJSCode' id="js"></td-code-editor>  
      </md-tab>
      <md-tab label="HTML">
        <td-code-editor automaticLayout="true" style="height: 200px" theme="vs" flex language="HTML" [(ngModel)]='frontEndHTMLCode' id="html"></td-code-editor>  
      </md-tab>
      <md-tab label="CSS">
        <td-code-editor #editor style="height: 200px" theme="vs" flex language="css" [(ngModel)]='frontEndCSSCode' id="css"></td-code-editor>  
      </md-tab>
    </md-tab-group>
    <button md-raised-button color="accent" (click)="editor.layout()" class="text-upper">SetLayout</button>
```
-[ ] npm run serve
-[ ] ng serve
-[ ] In browser go to: http://localhost:4200/#/components/code-editor
-[ ] See Tab group with Editors
-[ ] See Editor shows up in first two tabs and not last tab
-[ ] On last tab click SETLAYOUT button
-[ ] See Editor Show up

### CODE REVIEW

File: src/code-editor/code-editor.component.ts [hunk 1]
```diff
@@ -558,6 +589,7 @@ export class TdCodeEditorComponent implements OnInit, AfterViewInit, ControlValu
         value: this._value,
         language: this.language,
         theme: this._theme,
+        automaticLayout: this._automaticLayout,
```
Comment (emoralesb05) [inline src/code-editor/code-editor.component.ts:592]: Could this be updated on the fly after its initialized?

Comment (jeremysmartt) [reply src/code-editor/code-editor.component.ts:592]: no, I tried to call the update method in monaco for it and it didn't take.  So you have to destroy the instance and then recreate it.  So I was going to add a function for that, but then thought, why would you want to be toggling this setting on and off anyway, so left it out and it is only a one time thing at creation, just like with what monaco is doing.

Comment (emoralesb05) [reply src/code-editor/code-editor.component.ts:592]: cool, if thats the case should we make this an input without value?

```
<td-code-editor automaticLayout style="height: 200px" theme="vs" flex language="javascript" [(ngModel)]='frontEndJSCode' id="js"></td-code-editor> 
```

So its either there or not 😄  and it remains that way

Comment (jeremysmartt) [reply src/code-editor/code-editor.component.ts:592]: updated

**Comment by emoralesb05** [comment]
LGTM!

**Pull request merged**